client/X11/xf_monitor.c

```diff
@@ -285,36 +284,36 @@ BOOL xf_detect_monitors(xfContext* xfc, UINT32* pMaxWidth, UINT32* pMaxHeight)
 			/* does the same as gdk_rectangle_union */
 			int destX = MIN(vX, settings->MonitorDefArray[i].x);
 			int destY = MIN(vY, settings->MonitorDefArray[i].y);
-			int destWidth = MAX(vX + vWidth, settings->MonitorDefArray[i].x + settings->MonitorDefArray[i].width) - destX;
-			int destHeight = MAX(vY + vHeight, settings->MonitorDefArray[i].y + settings->MonitorDefArray[i].height) - destY;
+			int destR = MAX(vR, settings->MonitorDefArray[i].x + settings->MonitorDefArray[i].width);
+			int destB = MAX(vB, settings->MonitorDefArray[i].y + settings->MonitorDefArray[i].height);
 
 			if (vX != destX)
 				xfc->fullscreenMonitors.left = settings->MonitorDefArray[i].orig_screen;
 			if (vY != destY)
 				xfc->fullscreenMonitors.top = settings->MonitorDefArray[i].orig_screen;
-			if (vWidth != destWidth)
+			if (vR != destR)
 				xfc->fullscreenMonitors.right = settings->MonitorDefArray[i].orig_screen;
-			if (vHeight != destHeight)
+			if (vB != destB)
 				xfc->fullscreenMonitors.bottom = settings->MonitorDefArray[i].orig_screen;
 
 			vX = destX;
 			vY = destY;
-			vWidth = destWidth;
-			vHeight = destHeight;
+			vR = destR;
+			vB = destB;
 		}
 
 		settings->DesktopPosX = vX;
 		settings->DesktopPosY = vY;
 
 		vscreen->area.left = 0;
-		vscreen->area.right = vWidth - 1;
+		vscreen->area.right = vR - vX - 1;
 		vscreen->area.top = 0;
-		vscreen->area.bottom = vHeight - 1;
+		vscreen->area.bottom = vB - vY - 1;
 
 		if (settings->Workarea)
 		{
 			vscreen->area.top = xfc->workArea.y;
-			vscreen->area.bottom = (vHeight - (vHeight - (xfc->workArea.height + xfc->workArea.y))) - 1;
```

Comment:
Your reduction of the calculation looks good to me!
I believe my intent here was to show the operation in(what I considered at the time)a logical format. I imagine that when I worked out what to do here, I took the steps explicitly as I had them and did not realize that  was taking extra unnecessary steps to get the values that I wanted.